### PR TITLE
Update developer documentation

### PR DESCRIPTION
**Pull request by karliss**
**Detailed description**

* Create separate section for developer documentation - since we are planning to expand both user documentation and dev documentation, makes sens to have separate section for it. I left build instructions at top level as that is somewhat commonly done by users and packagers as well.
* Add first time contribution guide

**Test plan (required)**

I built the documentation locally and made sure the new links work and there are no warnings related to changes.

**Closing issues**

### CODE REVIEW

File: docs/source/dev-doc/first-time.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+* Come up with a plan for things that need to be done and discuss it in GitH	ub issue. Just because something was mentioned in issue description doesn't necessarily means that it is still relevant or good feature.
+* Check if any of the existing code implements similar behavior in the same widget or similar widgets. If you do copy an existing code consider why it did things the way it did, the same factors might not apply in your case or the old code is simply bad. Still it's a way of moving forwards if you otherwise wouldn't know how to approach it.
+* When code changes are done and you are ready for PR, please follow the PR template.
+
+"Test plan (required)" really means that. If those are GUI changes make a screenshot.
+Make a basic list of steps for checking that changes are working as expected. This also a good point to consider any potential edge cases or different kind of interesting inputs if you didn't already do it while writing code. Actually perform the steps you described when making the PR even if they seem trivial and you did them during development, it helps catching any makes done during final cleanup and making sure you didn't forget anything.
```
Comment (ITAYC0HEN) [inline docs/source/dev-doc/first-time.rst]: There is no `*` here, thus no new bullet point was opened. Is it intentionally?
```suggestion
* In the Pull Request template you will be required to add a "Test plan". For example, if you performed GUI changes, demonstrate it by posting a screenshot.
Make a basic list of steps for checking that changes are working as expected. This is also a good point to consider any potential edge-cases or different kinds of interesting inputs if you didn't already do it while writing the code. Perform the steps you described when making the PR even if they seem trivial and you did them during development, it helps to catch any mistakes done during the final cleanup and making sure you didn't forget anything.
```

File: docs/source/dev-doc.rst [hunk 2]
```diff
@@ -163,15 +168,19 @@ Example:
...
 
 To connect a signal to a slot, this is the preferred syntax:
 
 .. code:: cpp
 
    connect(sender, &QObject::destroyed, this, &MyObject::objectDestroyed);
 
-The main reason is that this syntax allows the use of lambda functions.
+This syntax performs compile time type checkign and allows the use of lambda
+functions. Other approaches for connecting signals silently break at runtime.
```
Comment (ITAYC0HEN) [inline docs/source/dev-doc.rst]:
```suggestion
This syntax performs compile-time type checks and allows the use of lambda
functions. Other approaches for connecting signals silently break at runtime.
```

File: docs/source/dev-doc.rst [hunk 3]
```diff
@@ -213,3 +222,11 @@ In order to update one submodule individually, use the following code:
    cd ..
    git add radare2
    git commit -m "Update radare2 submodule"
+
+
+Usefull things to learn about Qt
```
Comment (ITAYC0HEN) [inline docs/source/dev-doc.rst]:
```suggestion
Useful resources to learn more about Qt development
```

File: docs/source/dev-doc.rst [hunk 4]
```diff
@@ -213,3 +222,11 @@ In order to update one submodule individually, use the following code:
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* `Signals & Slots <https://doc.qt.io/qt-5/signalsandslots.html>`__
+* `Model/View Programming <https://doc.qt.io/qt-5/model-view-programming.html>`__ read this if you are going to work with list or table like widgtets
```
Comment (ITAYC0HEN) [inline docs/source/dev-doc.rst]:
```suggestion
* `Model/View Programming <https://doc.qt.io/qt-5/model-view-programming.html>`__ - read this if you are going to work with list or table-like widgets
```

File: docs/source/dev-doc/first-time.rst [hunk 5]
```diff
@@ -0,0 +1,15 @@
+First time contributing
+=======================
+
+
+* Start by reading :doc:`../dev-doc`
+* See :doc:`../building` for compilation instructions
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/dev-doc/first-time.rst]: Done. Although I think it's slightly too verbose.

File: docs/source/dev-doc/first-time.rst [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+* Get the build working
+* Run cutter and get familiar with how the functionality you want to modify works currently
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/dev-doc/first-time.rst]: Done, added a sentence about reproducing a bug.

File: docs/source/dev-doc/first-time.rst [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+* Find the the source files implementing it (practice searching code in your editor, this is an important skill)
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/dev-doc/first-time.rst]: Looks good. I also wanted to mention searching for messages or other strings.

File: docs/source/dev-doc/first-time.rst [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+* Come up with a plan for things that need to be done and discuss it in GitH	ub issue. Just because something was mentioned in issue description doesn't necessarily means that it is still relevant or good feature.
+* Check if any of the existing code implements similar behavior in the same widget or similar widgets. If you do copy an existing code consider why it did things the way it did, the same factors might not apply in your case or the old code is simply bad. Still it's a way of moving forwards if you otherwise wouldn't know how to approach it.
+* When code changes are done and you are ready for PR, please follow the PR template.
```
[the remark this answers is not in the document]
Comment (karliss) [reply docs/source/dev-doc/first-time.rst]: Done

File: docs/source/dev-doc/first-time.rst (hunk 1 above)
Comment (karliss) [reply docs/source/dev-doc/first-time.rst]: Done

**Pull request merged**